### PR TITLE
fix(tianmu): fix bug:'on duplicate key update' scenario, incorrect result occurs. (#502)

### LINKED ISSUE #502: bug:  incorrect result set.Check that ON DUPLICATE KEY UPDATE，A query result error occurs
### Describe the problem

```
CREATE TABLE t1(a1 INTEGER PRIMARY KEY, b1 INTEGER)engine=tianmu;
CREATE TABLE t2(a2 INTEGER PRIMARY KEY, b2 INTEGER)engine=tianmu;
CREATE VIEW v AS SELECT * FROM t1 JOIN t2 ON TRUE;
INSERT INTO v(a1,b1) VALUES (11, 0) ON DUPLICATE KEY UPDATE b1=b1+1;
INSERT INTO v(a1,b1) VALUES (11, 0) ON DUPLICATE KEY UPDATE b1=b1+1;
INSERT INTO v(a2,b2) VALUES (21, 0) ON DUPLICATE KEY UPDATE b2=b2+1;
INSERT INTO v(a2,b2) VALUES (21, 0) ON DUPLICATE KEY UPDATE b2=b2+1;
SELECT * FROM v;

#Incorrect test results
mysql> SELECT * FROM v;
+----+------+----+------+
| a1 | b1   | a2 | b2   |
+----+------+----+------+
| 11 |    0 | 21 |    0 |
+----+------+----+------+
```

### Expected behavior

# Correct test results（note:Innodb test verification, test results are as follows）
```
mysql> SELECT * FROM v;
+----+------+----+------+
| a1 | b1   | a2 | b2   |
+----+------+----+------+
| 11 |    1 | 21 |    1 |
+----+------+----+------+
```

### How To Reproduce

```
CREATE TABLE t1(a1 INTEGER PRIMARY KEY, b1 INTEGER)engine=tianmu;
CREATE TABLE t2(a2 INTEGER PRIMARY KEY, b2 INTEGER)engine=tianmu;
CREATE VIEW v AS SELECT * FROM t1 JOIN t2 ON TRUE;
INSERT INTO v(a1,b1) VALUES (11, 0) ON DUPLICATE KEY UPDATE b1=b1+1;
INSERT INTO v(a1,b1) VALUES (11, 0) ON DUPLICATE KEY UPDATE b1=b1+1;
INSERT INTO v(a2,b2) VALUES (21, 0) ON DUPLICATE KEY UPDATE b2=b2+1;
INSERT INTO v(a2,b2) VALUES (21, 0) ON DUPLICATE KEY UPDATE b2=b2+1;
SELECT * FROM v;
```

### Environment

```
[root@dev bin]# ./mysqld --version
./mysqld  Ver 5.7.36_v1.0.0_beta-StoneDB for Linux on x86_64 (build-)
build information as follow: 
        Repository address: https://github.com/stoneatom/stonedb.git:stonedb-5.7
        Branch name: stonedb-5.7
        Last commit ID: eed32f6
        Last commit time: Date:   Wed Aug 3 11:19:48 2022 +0800
        Build time: Date: Wed Aug  3 13:01:31 CST 2022
```

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

### PR DESCRIPTION
**Pull request by lujiashun**
## Summary about this PR

Issue Number: close #502

[summary]
1 the root reason is because of the variable ‘tianmu_insert_delayed’,set it off; 
2 in debug version, it crashes, fix it;
3 add mtr case for this issue;

## Tests Check List

- [ ] Unit test
- [X] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [X] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: mysql-test/suite/tianmu/t/issue502-2.test [hunk 1]
```diff
@@ -0,0 +1,14 @@
+create database test_tmp;
```
Comment (RingsC) [inline mysql-test/suite/tianmu/t/issue502-2.test:2]: Pls, using source to include have_tianmu.inc in. Thanks.

Comment (RingsC) [reply mysql-test/suite/tianmu/t/issue502-2.test:2]: And, adds tests for string and unicode string.

**Review by RingsC — CHANGES_REQUESTED** [review]

**Review by RingsC — APPROVED** [review]
LGTM

**Review by isredstar — APPROVED** [review]
LGTM

**Pull request merged**